wicket-core/src/main/java/org/apache/wicket/resource/FileSystemResourceReference.java

```diff
@@ -0,0 +1,181 @@
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.wicket.resource;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.nio.file.FileSystem;
+import java.nio.file.FileSystems;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.wicket.request.resource.IResource;
+import org.apache.wicket.request.resource.ResourceReference;
+import org.apache.wicket.util.lang.Args;
+
+/**
+ * This resource reference is used to provide a reference to a resource based on Java NIO FileSystem
+ * API.<br>
+ * <br>
+ * To implement a mime type detection refer to the documentation of
+ * {@link java.nio.file.Files#probeContentType(Path)} and provide an implementation for
+ * java.nio.file.spi.FileTypeDetector in the META-INF/services folder for jars or in the
+ * /WEB-INF/classes/META-INF/services folder for webapps<br>
+ * <br>
+ * You can optionally override {@link #getMimeType()} to provide an inline mime type detection,
+ * which is preferred to the default detection.<br>
+ * <br>
+ * Example:
+ * 
+ * <pre>
+ * <code>
+ * Path path = FileSystemResourceReference.getPath(URI.create("jar:file:///folder/file.zip!/folderInZip/video.mp4"));
+ * add(new Video("video", new FileSystemResourceReference(path)));
+ * </code>
+ * </pre>
+ * 
+ * @author Tobias Soloschenko
+ */
+public class FileSystemResourceReference extends ResourceReference
+{
+	private static final long serialVersionUID = 1L;
+
+	private final Path path;
+
+	private static Map<URI, FileSystem> fileSystemURIs = new HashMap<URI, FileSystem>();
+
+	/**
+	 * Creates a file system resource reference based on the given path
+	 * 
+	 * @param name
+	 *            the name of the resource reference to expose data
+	 * @param path
+	 *            the path to create the resource reference
+	 */
+	public FileSystemResourceReference(String name, Path path)
+	{
+		super(Args.notNull(name, "name"));
+		Args.notNull(path, "path");
+		this.path = path;
+	}
+
+	/**
+	 * Creates a file system resource reference based on the given path
+	 * 
+	 * @param path
+	 *            the path to create the resource reference (the name is used to expose the data)
+	 */
+	public FileSystemResourceReference(Path path)
+	{
+		super(Args.notNull(path, "path").getFileName().toString());
```

Comment:
This will fail if you have two files with the same file names, e.g. `/one/a` and `/two/a`. Both will try to use `a` as a name.
And since the name is used for the url for this resource reference we cannot use the absolute path. IMO this constructor should be removed in favor of `(public FileSystemResourceReference(String name, Path path))` above.